python/cudf/cudf/core/column/column.py

```diff
@@ -558,14 +560,14 @@ def __setitem__(self, key, value):
                 if is_scalar(value):
                     input = self
                     if is_categorical_dtype(self.dtype):
-                        input = self.codes
+                        input = self.children[0]
```

Comment:
Do you need this to be mask aware? If so we should switch it back to `self.codes`